client/student/app/lecture/lecture.factory.js

```diff
@@ -35,14 +36,23 @@
             } else {
                 $location.path('/');
             }
+            
+            function onSuccess(value, responseHeaders) {
+                console.log('Success', value);
+            }
+            
+            function onFail(httpResponse) {
+                console.log('Error');
+            }
         }
         
-        function onSuccess(value, responseHeaders) {
-            console.log('success', value);
-        }
-        
-        function onFail(httpResponse) {
-            console.log('error: ', httpResponse.status, httpResponse.statusText);
+        function getComments(onSuccess, onFail) {
+            if (angular.isDefined(lectureId)) {
+                lectureService.retrieveComments
+                    .get({id: lectureId}, onSuccess, onFail);
+            } else {
+                $location.path('/');
```

Comment:
I'm a bit confused by this. Will it go to the main page when lectureId is not defined?